### PR TITLE
[SPARK-22357][CORE][FOLLOWUP] SparkContext.binaryFiles ignore minPartitions parameter

### PR DESCRIPTION
**Pull request by srowen**
## What changes were proposed in this pull request?

This adds a test following https://github.com/apache/spark/pull/21638

## How was this patch tested?

Existing tests and new test.

### CODE REVIEW

**Review by imatiach-msft — APPROVED** [review]
nice test!

File: core/src/test/scala/org/apache/spark/FileSuite.scala [hunk 1]
```diff
@@ -299,6 +301,25 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
...
+    val tempDir = Utils.createTempDir()
+    val tempDirPath = tempDir.getAbsolutePath
+
+    for (i <- 0 until 8) {
+      val tempFile = new File(tempDir, s"part-0000$i")
+      Files.write("someline1 in file1\nsomeline2 in file1\nsomeline3 in file1", tempFile,
+        StandardCharsets.UTF_8)
+    }
+
+    assert(sc.binaryFiles(tempDirPath, minPartitions = 1).getNumPartitions === 1)
```
Comment (imatiach-msft) [inline core/src/test/scala/org/apache/spark/FileSuite.scala]: nitpick: maybe put these three asserts in a loop

File: core/src/test/scala/org/apache/spark/FileSuite.scala [hunk 2]
```diff
@@ -299,6 +301,25 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
     }
   }
 
+  test("SPARK-22357 test binaryFiles minPartitions") {
+    sc = new SparkContext(new SparkConf().setAppName("test").setMaster("local")
+      .set("spark.files.openCostInBytes", "0")
```
Comment (imatiach-msft) [inline core/src/test/scala/org/apache/spark/FileSuite.scala]: why is this setting needed: spark.files.openCostInBytes

**Review by imatiach-msft — APPROVED** [review]
comments

File: core/src/test/scala/org/apache/spark/FileSuite.scala (hunk 1 above)
Comment (srowen) [reply core/src/test/scala/org/apache/spark/FileSuite.scala]: OK, sure

File: core/src/test/scala/org/apache/spark/FileSuite.scala (hunk 2 above)
Comment (srowen) [reply core/src/test/scala/org/apache/spark/FileSuite.scala]: This removes its effect in the section of code we're really trying to test:

```
def setMinPartitions(sc: SparkContext, context: JobContext, minPartitions: Int) {
    val defaultMaxSplitBytes = sc.getConf.get(config.FILES_MAX_PARTITION_BYTES)
    val openCostInBytes = sc.getConf.get(config.FILES_OPEN_COST_IN_BYTES)
    val defaultParallelism = Math.max(sc.defaultParallelism, minPartitions)
    val files = listStatus(context).asScala
    val totalBytes = files.filterNot(_.isDirectory).map(_.getLen + openCostInBytes).sum
    val bytesPerCore = totalBytes / defaultParallelism
    val maxSplitSize = Math.min(defaultMaxSplitBytes, Math.max(openCostInBytes, bytesPerCore))
    super.setMaxSplitSize(maxSplitSize)
  }
```

Comment (imatiach-msft) [reply core/src/test/scala/org/apache/spark/FileSuite.scala]: ah, I see, thanks for pointing that out!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master/2.4

**Pull request closed without merging**